test/test_operators.jl

```diff
@@ -2,323 +2,394 @@ using Test
-    s; n, boundary="Dirichlet", truncate_kwargs=(; cutoff=1e-10)
-  )
-  forward_shift_pbc = forward_shift_op(
-    s; n, boundary="Periodic", truncate_kwargs=(; cutoff=1e-10)
-  )
-  backward_shift_pbc = backward_shift_op(
-    s; n, boundary="Periodic", truncate_kwargs=(; cutoff=1e-10)
-  )
-  forward_shift_neumann = forward_shift_op(
-    s; n, boundary="Neumann", truncate_kwargs=(; cutoff=1e-10)
-  )
-  backward_shift_neumann = backward_shift_op(
-    s; n, boundary="Neumann", truncate_kwargs=(; cutoff=1e-10)
-  )
-
-  ψ_fx_pshift_dirichlet = operate(forward_shift_dirichlet, ψ_fx; cutoff=1e-12)
-  ψ_fx_mshift_dirichlet = operate(backward_shift_dirichlet, ψ_fx; cutoff=1e-12)
-  ψ_fx_pshift_pbc = operate(forward_shift_pbc, ψ_fx; cutoff=1e-12)
-  ψ_fx_mshift_pbc = operate(backward_shift_pbc, ψ_fx; cutoff=1e-12)
-  ψ_fx_pshift_neumann = operate(forward_shift_neumann, ψ_fx; cutoff=1e-12)
-  ψ_fx_mshift_neumann = operate(backward_shift_neumann, ψ_fx; cutoff=1e-12)
-
-  for x in xs
-    if x + 2.0 * delta < 1
-      fx_xplus = calculate_fx(ψ_fx, x + 2.0 * delta)
-      @test fx_xplus ≈ calculate_fx(ψ_fx_pshift_dirichlet, x) atol = 1e-8
-      @test fx_xplus ≈ calculate_fx(ψ_fx_pshift_pbc, x) atol = 1e-8
-      @test fx_xplus ≈ calculate_fx(ψ_fx_pshift_neumann, x) atol = 1e-8
-    elseif x == 1.0 - 2.0 * delta || x == 1.0 - delta
-      @test calculate_fx(ψ_fx_pshift_dirichlet, x; alg="exact") ≈ 0.0 atol = 1e-8
-      @test calculate_fx(ψ_fx_pshift_pbc, x) ≈ calculate_fx(ψ_fx, x + 2.0 * delta - 1.0) atol =
-        1e-8
-      @test calculate_fx(ψ_fx_pshift_neumann, x) ≈ calculate_fx(ψ_fx, x) atol = 1e-8
-    end
+  @testset "test boundary operator in 1D on Tree" begin
+    g = named_comb_tree((2, 3))
+    L = nv(g)
+    delta = 2.0^(-1.0 * L)
+    lastDigit = 1 - delta
+    s = continuous_siteinds(g)
+
+    xs = [0.0, delta, 0.25, 0.5, 0.625, 0.875, 1.0 - delta]
+    ψ_fx = poly_itn(s, [1.0, 0.5, 0.25])
 
-    if x - 2.0 * delta >= 0.0
-      fx_xminus = calculate_fx(ψ_fx, x - 2.0 * delta)
-      @test fx_xminus ≈ calculate_fx(ψ_fx_mshift_dirichlet, x) atol = 1e-8
-      @test fx_xminus ≈ calculate_fx(ψ_fx_mshift_pbc, x) atol = 1e-8
-      @test fx_xminus ≈ calculate_fx(ψ_fx_mshift_neumann, x) atol = 1e-8
-    else
-      @test calculate_fx(ψ_fx_mshift_dirichlet, x) ≈ 0.0 atol = 1e-8
-      @test calculate_fx(ψ_fx_mshift_pbc, x) ≈ calculate_fx(ψ_fx, x - 2.0 * delta + 1.0) atol =
-        1e-8
-      @test calculate_fx(ψ_fx_mshift_neumann, x) ≈ calculate_fx(ψ_fx, x) atol = 1e-8
+    Zo = map_to_zero_operator(s, [0, lastDigit])
+
+    @testset "corner boundary test" begin
+      for p1 in [0, lastDigit]
+        p = ttn(itensornetwork(delta_x(s, p1)))
+        @test inner(p', Zo, p) ≈ 0.0
+      end
+    end
+    @testset "boundary apply" begin
+      maxdim, cutoff = 10, 0e-16
+      #ϕ_fx = operate(Zo, ψ_fx; cutoff, maxdim, normalize=false)
+      ϕ_fx = map_to_zeros(ψ_fx, [0, lastDigit]; cutoff, maxdim)
+      for x in [0, lastDigit]
```

Comment:
Maybe just 

```
for x in xs
  val = real(calculate_fx(ϕ_fx, x; alg="exact"))
  if x in [0, lastdigit]
     @test val ≈ 0.0 atol = 1e-8
  else
     @test !(val ≈ 0.0 atol = 1e-8)
  end
end
```
to avoid a double for loop